src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -111,23 +100,36 @@ def odesmooth(self, means, covs, times, **kwargs):
 
         Parameters
         ----------
-        means
-        covs
+        filter_solution: ODESolution
 
         Returns
         -------
-
+        smoothed_solution: ODESolution
         """
-        means, covs = self.redo_preconditioning(means, covs)
-        means, covs = self.gfilt.smooth(means, covs, times, **kwargs)
-        return self.undo_preconditioning(means, covs)
+        ivp_filter_posterior = filter_solution._state_posterior
+        ivp_smoother_posterior = self.gfilt.smooth(ivp_filter_posterior, **kwargs)
+
+        smoothed_solution = ODESolution(
+            times=ivp_smoother_posterior.locations,
+            rvs=ivp_smoother_posterior.state_rvs,
+            solver=filter_solution._solver,
+        )
+
+        return smoothed_solution
 
     def undo_preconditioning(self, means, covs):
         ipre = self.gfilt.dynamicmodel.invprecond
         newmeans = np.array([ipre @ mean for mean in means])
         newcovs = np.array([ipre @ cov @ ipre.T for cov in covs])
         return newmeans, newcovs
 
+    def undo_preconditioning_rv(self, rv):
```

Comment:
make sure that "undo_preconditioning" is not used anywhere, except for here, then
merge this with undo_preconditioning